src/Web/Rollbar/Types.hs

```diff
@@ -1,37 +1,141 @@
+    , eventLevel
+    , eventMessage
+    , eventTitle
+    , eventUUID
+    ) where
 
-import Control.Lens.TH
-import Data.Aeson (Value)
+import Control.Lens.TH (makeClassy, makeLenses)
+import Data.Aeson (ToJSON(..), Value(String))
+import Data.Ord (Ord)
+import Data.Text (Text)
 
+-- | Grants access to the Rollbar API. Can be retrieved from
+-- @https://rollbar.com/<org>/<project>/settings/access_tokens/@
+newtype AccessToken = AccessToken
+    { unAccessToken :: Text
+    }
+
+instance ToJSON AccessToken where
+    toJSON = toJSON . unAccessToken
+
+-- | Environment in which the code runs, e.g. @production@,
+-- @staging@, @development@, etc.
+newtype Environment = Environment
+    { unEnvironment :: Text
+    }
+
+instance ToJSON Environment where
+    toJSON = toJSON . unEnvironment
+
+-- | Version of the code that is running.
+--
+-- Max length: 40 characters
+--
+-- Rollbar understands these formats:
+--
+--      * semantic version, e.g. @"2.1.12"@
+--      * integer, e.g. @"42"@
+--      * Git SHA, e.g. @"3da541559918a808c2402bba5012f6c60b27661c"@
+newtype CodeVersion = CodeVersion
+    { unCodeVersion :: Text
+    }
+
+instance ToJSON CodeVersion where
+    toJSON = toJSON . unCodeVersion
+
+-- | Server hostname, indexed by Rollbar. Can be used to describe the service
+-- name, e.g. @apiary@, @gourd@, @trellis@, etc.
+--
+-- Rollbar allows search by prefix, e.g. searching @apiary@ will find both
+-- @apiary-us-east-1@ and @apiary-us-west-2@.
+newtype Host = Host
+    { unHost :: Text
+    }
+
+instance ToJSON Host where
+    toJSON = toJSON . unHost
+
+-- | Configuration used to post events to the Rollbar API.
 data RollbarCfg = RollbarCfg
-    { _rollbarCfgToken :: String
-    , _rollbarCfgEnvironment :: String
-    , _rollbarCfgMute :: Bool
+    { _rollbarCfgToken :: !AccessToken
+    , _rollbarCfgEnvironment :: !Environment
+    , _rollbarCfgHost :: !(Maybe Host)
+    , _rollbarCfgCodeVersion :: !(Maybe CodeVersion)
+    , _rollbarCfgMute :: !Bool
     }
 
+-- | Event that is posted to Rollbar.
+-- See: https://docs.rollbar.com/reference#section-data-format
 data Event = Event
-    { _eventLevel :: EventLevel
-    , _eventUUID :: Maybe String
-    , _eventTitle :: String
-    , _eventMessage :: String
-    , _eventData :: Maybe Value
-    } deriving (Show)
+    { -- | Severity level
+      _eventLevel :: !EventLevel
+      -- | Uniquely identifies this occurrence.
+      -- We recommend using a UUID4 (16 random bytes).
+      --
+      -- Max length: 36 characters
+      --
+      -- The UUID space is unique to each project and can be used to look up an
+      -- occurrence later.
+      -- The UUID is also used to detect duplicate requests. If you send the
+      -- same UUID in two payloads, the second one will be discarded.
+      -- While optional, it is recommended that all clients generate and provide
+      -- this field.
+      -- /TODO: Pick better type and generate this internally./
+    , _eventUUID :: !(Maybe Text)
```

Comment:
**TODO:** We should really generate that during the report call. Also: We should generate our own `timestamp`.

Reply:
Nothing! I think we should do that that’s why I left a comment: `/TODO: Pick better type and generate this internally./` I plan to address that in Phase 2: https://collegevine.atlassian.net/browse/PE-1247